lib/IRGen/GenClass.cpp

```diff
@@ -317,6 +317,12 @@ namespace {
           NumInherited = Elements.size();
         }
       }
+      
+      // If this class was imported from another module, assume that we may
+      // not know its exact layout.
+      if (theClass->getModuleContext() != IGM.getSwiftModule()) {
```

Comment:
What if the superclass is from a different module? I think you need to change the isResilient() test above

Reply:
This function is called recursively for the entire superclass chain. Setting this when we see the base class sticks when we get to the subclass fields.